src/Authentication/Authentication/custom/common/Permissions.ps1

```diff
@@ -1,36 +1,49 @@
 # ------------------------------------------------------------------------------
 #  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
 # ------------------------------------------------------------------------------
-
-
-$permissionsPulledFromMgGraphRequest = $null
+Update-FormatData -PrependPath $PSScriptRoot\..\..\Microsoft.Graph.Authentication.format.ps1xml
 
 function Permissions_GetPermissionsData {
+    param (
+        [bool] $online
+    )
 
     $permissions_MsGraphServicePrincipal = $null
+    $requestError = $null
+    $fromInvokeMgGraphRequest = $false
 
     # 2. Making a REST request to MS Graph
-    if ($null -eq $permissions_MsGraphServicePrincipal){
+
+    if (($null -eq $permissions_MsGraphServicePrincipal) -or ($null -ne $permissions_MsGraphServicePrincipal -and $fromInvokeMgGraphRequest -eq $false)) {
```

Comment:
Need to check `$online` and do this if `$online` is specified in all cases -- be careful though not to overwrite existing sp data until after you're successful!